dashboard/app/models/levels/level.rb

```diff
@@ -319,7 +319,7 @@ def self.write_custom_levels
   end
 
   def should_write_custom_level_file?
-    changed = changed? || (level_concept_difficulty && level_concept_difficulty.changed?)
+    changed = saved_changes? || (level_concept_difficulty && level_concept_difficulty.saved_changes?)
```

Comment:
It looks like you've limited this change to places where `changed?` is called after save, which is good. However, my one concern with this change overall is any places that the same code you are touching might also be called outside of an after save callback.

On this particular line in this PR, `write_custom_level_file` and `should_write_custom_level_file?` can also get called via the following codepaths:
* https://github.com/code-dot-org/code-dot-org/blob/49a7cfd526e0c0a4ad610b85534a810ba45b565f/dashboard/app/models/levels/level.rb#L315-L317
* https://github.com/code-dot-org/code-dot-org/blob/961529db99e7ed1a32c8b76cbd0bbea2f4abf778/bin/oneoff/fix_xml_newlines.rb#L125

The former looks like it is unused, so I would suggest removing it in this PR. The latter looks like it might be broken by this change, but I'm not sure if it needs to still work or not. It was introduced in https://github.com/code-dot-org/code-dot-org/pull/24353 so perhaps you are the best person to decide what to do with it? If it is no longer needed, then I'd recommend removing the entire script.

For additional context, when used outside of an after save callback, the `changed?` method appears to tell us if there are any pending changes to the object which have not yet been saved: https://github.com/code-dot-org/code-dot-org/blob/ab81d45729ed8e2e5581340591af7eb8cdfeb623/dashboard/app/models/script_level.rb#L127 I haven't checked the latest docs to see if this is changing any time soon, though.

Reply:
Thanks for doing this legwork! I hadn't noticed these at all, but I think you're right that we can remove them. The XML fix was a onetime fix that it doesn't look like we need anymore.

I'll open up a separate PR to remove both those codepaths.

My understanding is that the behavior or `changed?` outside of an after callback will not be changing, so the rest of our codebase should be fine.